docs/get-details/dapps/smart-contracts/apps/index.md

```diff
@@ -452,7 +452,73 @@ Additionally, when validating transactions, using relative position of transacti
 
 Since TEAL 6, all created assets and apps are available to be accessed by application calls in the same group which allows more dynamic behavior. For example an application can be created via Inner Transaction then funded immediately in the same transaction group since we have access to the created application id and address.   
 
-# Modifying state in smart contract
+
+# Smart Contract Storage 
+Smart Contracts have three different types of storage: local storage, global storage, and box storage. 
+
+Global state and boxes are associated with the app itself, whereas local state is associated with each user account that opts into the application. Global and local storage are Key/Value pairs that are limited to 128 bytes per pair. Boxes are keyed storage segments up to 32kb of data per box.
+
+Each storage option’s properties are described below. 
+
+**Global state for app a:**
+
+* Allocation: 
+    * Can include between 0 and 64 key/value pairs and a total of 8K of memory to share among them.
+    * The amount of global storage is allocated in k/v units, and determined at contract creation. This cannot be edited later. 
+    * The contract creator address is responsible for funding the global storage (by an increase to their minimum balance requirement, see below). 
+* Reading: 
+    * Can be read by any app call that has specified app a’s ID in its foreign apps array.
+    * Can be read on-chain using the k/v pairs defined (from off-chain, can be read using goal or APIs + SDKs). 
+* Writing: 
+    * Can only be written by app a. 
+* Deletion: 
+    * Is deleted when app a is deleted. Cannot otherwise be deallocated (though of course the contents can be cleared by app a, but this does not change the minimum balance requirement). 
+
+**Local state for account x for app a:** 
+
+* Allocation: 
+    * Is allocated when account x opts in to app a (submits a transaction to opt-in to app a).
+    * Can include between 0 and 16 key/value pairs and a total of 2KB of memory to share among them. 
+    * The amount of local storage is allocated in k/v units, and determined at contract creation. This cannot be edited later. 
+    * The opted-in user address is responsible for funding the local storage (by an increase to their minimum balance). 
+* Reading: 
+    * Can be read by any app call that has app x in its foreign apps array and account x in its foreign accounts array. 
+    * Can be read on-chain using the k/v pairs defined (from off-chain, can be read using goal and the SDKs). 
+* Writing: 
+    * Is editable only by app a, but is delete-able by app a or the user x (using a ClearState call, see below). 
+* Deletion: 
+    * Deleting an app does not affect its local storage. 
```

Comment:
??

Reply:
wouldnt the local storage disappear? the min balance requirement would remain until the account opted out but id thought the local storage values would be inaccessible